core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/FixedRateSamplingTelemetryProcessor.java

```diff
@@ -0,0 +1,175 @@
+package com.microsoft.applicationinsights.internal.channel.samplingV2;
+
+import com.microsoft.applicationinsights.agent.internal.common.StringUtils;
+import com.microsoft.applicationinsights.extensibility.TelemetryProcessor;
+import com.microsoft.applicationinsights.internal.annotation.BuiltInProcessor;
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import com.microsoft.applicationinsights.telemetry.SupportSampling;
+import com.microsoft.applicationinsights.telemetry.Telemetry;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Created by Dhaval Doshi Oct 2017
+ * This processor is used to Perform Sampling on User specified sampling rate
+ */
+@BuiltInProcessor("FixedRateSamplingTelemetryProcessor")
+public final class FixedRateSamplingTelemetryProcessor implements TelemetryProcessor {
+
+    private final String dependencyTelemetryName = "Dependency";
+    private static final String eventTelemetryName = "Event";
+    private static final String exceptionTelemetryName = "Exception";
+    private static final String pageViewTelemetryName = "PageView";
+    private static final String requestTelemetryName = "Request";
+    private static final String traceTelemetryName = "Trace";
+
+    private static final String listSeparator = ";";
+    private static Map<String, Class> allowedTypes;
+
+    private String excludedTypesString;
+    private Set<Class> excludedTypesHashSet;
+
+    private String includedTypesString;
+    private Set<Class> includedTypesHashSet;
+
+    public double samplingPercentage;
```

Comment:
@SergeyKanzhelev 

Addressing 1. There are getters and setters for this however they accept string as input rather than double. This is because our configuration to initialize processors require that. Its public because dot net had kept it public however I don't see it getting accessed directly without getters anywhere so probably can make this private.

Reply:
Addressing 2. Sure I will add this comment makes sense. 100/N should be format.